source/common/network/address_impl.cc

```diff
@@ -8,30 +8,86 @@
 namespace Network {
 namespace Address {
 
+Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss, socklen_t ss_len) {
+  if (ss_len != 0 && ss_len <= sizeof(sa_family_t)) {
+    throw EnvoyException(fmt::format("sockaddr_storage is too short: {}", ss_len));
+  }
+  switch (ss.ss_family) {
+  case AF_INET:
+    if (ss_len == 0 || ss_len == sizeof(sockaddr_in)) {
+      const struct sockaddr_in* sin = reinterpret_cast<const struct sockaddr_in*>(&ss);
+      // Only ASSERT-ing in debug builds because definition of sockaddr_storage and sockaddr_in
+      // requires them to share the same storage for the ss_family and sin_family fields.
+      ASSERT(AF_INET == sin->sin_family);
+      return InstanceConstSharedPtr(new Address::Ipv4Instance(sin));
```

Comment:
Can we use std::make_shared<> here and below for slight perf boost (I'm not sure if modern compilers do this automatically but might as well be explicit).